src/unix/stdpaths.cpp

```diff
@@ -167,6 +177,28 @@ wxString wxStandardPaths::GetExecutablePath() const
 
     if ( !exeStr.empty() )
         return exeStr;
+#elif defined(__FreeBSD__) && wxHAS_PROCSTAT
+    unsigned int n_proc;
+    char pathname[PATH_MAX];
+
+    auto *ps = procstat_open_sysctl();
+    if (!ps)
+        return wxStandardPathsBase::GetExecutablePath();
+
+    auto *procs = procstat_getprocs(ps, KERN_PROC_PID, getpid(), &n_proc);
+    if (!procs || n_proc != 1)
+        return wxStandardPathsBase::GetExecutablePath();
+
+    // procstat_freeprocs should be called before procstat_close
+    wxON_BLOCK_EXIT2(procstat_freeprocs, ps, procs);
+    wxON_BLOCK_EXIT1(procstat_close, ps);
```

Comment:
This line needs to be moved above (before `procs` initialization), otherwise `procstat_close()` won't be called if `procstat_getprocs()` fails and we return early.

Reply:
`procstat_close` needs to be called after `procstat_freeprocs` so that `ps` is still valid in the `procstat_freeprocs` call. I was thinking code execution order, but the destructors would be run while clearing the stack, so that would be lifo making the first code line the last to run, which would work.

Reply:
Not sure if I'm missing something, but yes, exactly: because the dtors run in the opposite order, you have to put the guard for `procstat_close()` *before* the one for `procstat_freeprocs()`.